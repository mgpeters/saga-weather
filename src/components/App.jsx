/**
 * ************************************
 *
 * @module  App.jsx
 * @author  mpeters
 * @date    09/30/2020
 * @description Weather application root
 * component
 *
 * ************************************
 */

import React, { Component } from 'react';
import { connect } from 'react-redux';

import * as actions from '../state/actions/actions';

const mapStateToProps = (store) => ({
  locationData: store.locationData,
});

const mapDispatchToProps = (dispatch) => ({
  fetchWeather: (location) => {
    dispatch(actions.getWeather(location));
  },
  updateLocation: () => {
    dispatch(actions.updateLocation());
  },
});

class App extends Component {
  componentDidMount() {
    this.props.fetchWeather('newyork');
    this.props.fetchWeather('miami');
    this.props.fetchWeather('losangeles');
  }

  render() {
    return (
      <div>
        <h1>WORKING</h1>
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
