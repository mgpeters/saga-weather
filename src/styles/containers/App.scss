@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@300&family=Zilla+Slab:ital@1&display=swap');

body {
  height: 100%;
}
.main-container {
  height: 100%;
  h1 {
    color: bisque;
  }
}
