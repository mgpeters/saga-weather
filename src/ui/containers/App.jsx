/**
 * ************************************
 *
 * @module  App.jsx
 * @author  mpeters
 * @date    09/30/2020
 * @description Weather application root
 * component
 *
 * ************************************
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Nav from './Nav.jsx';
import WeatherDisplay from './WeatherDisplay.jsx';

import * as actions from '../../state/actions/actions';

const mapStateToProps = (store, ownProps) => ({
  currentLocation: store.weather.currentLocation,
  currentPathname: ownProps.location.pathname,
});

const mapDispatchToProps = (dispatch) => ({
  updatePathname: (pathname) => {
    dispatch(actions.updatePathname(pathname));
  },
});
class App extends Component {
  componentDidMount() {
    console.log('App Mounted');
  }
  componentDidUpdate() {
    console.log('App Updated');
    if (this.props.currentLocation.name) {
      document.title = this.props.currentLocation.name;
    }
    // this.props.updatePathname(this.props.currentPathname);
  }

  render() {
    return (
      <section className="main-container">
        <Route path="/:location?" component={Nav} />
        <Switch>
          <Route path="/:location?" component={WeatherDisplay} history />
        </Switch>
      </section>
    );
  }
}

App.propTypes = {
  currentLocation: PropTypes.object,
  currentPathname: PropTypes.string,
  updatePathname: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
