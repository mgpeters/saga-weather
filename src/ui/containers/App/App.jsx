/**
 * ************************************
 *
 * @module  App.jsx
 * @author  mpeters
 * @date    09/30/2020
 * @description Weather application root
 * component
 *
 * ************************************
 */

import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import styles from '../../../styles/containers/App.scss';
import * as actions from '../../../state/actions/actions';

const mapStateToProps = (store) => ({
  currentLocation: store.currentLocation,
  locations: store.locations,
  locationData: store.locationData,
});

const mapDispatchToProps = (dispatch) => ({
  fetchWeather: (location) => {
    dispatch(actions.getWeather(location));
  },
  updateLocation: () => {
    dispatch(actions.updateLocation());
  },
});

class App extends Component {
  componentDidMount() {
    this.props.fetchWeather('newyork');
  }

  componentDidUpdate() {
    document.title = this.props.currentLocation;
  }

  render() {
    return (
      <div>
        <h1>WORKING</h1>
      </div>
    );
  }
}

App.propTypes = {
  currentLocation: PropTypes.string,
  locations: PropTypes.arrayOf(PropTypes.string),
  locationData: PropTypes.object,
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
