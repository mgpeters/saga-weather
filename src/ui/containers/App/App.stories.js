import React from 'react';
import Provider from '../../../util/Provider';

import store from '../../../state/store';

import App from './App.jsx';

const withProvider = (story) => <Provider store={store}>{story()}</Provider>;

export default {
  title: 'Containers/App',
  component: App,
  decorators: [withProvider],
};

const Template = (args) => <App {...args} />;

export const LoggedIn = Template.bind({});
// LoggedIn.args = {
//   ...HeaderStories.LoggedIn.args,
// };

export const LoggedOut = Template.bind({});
// LoggedOut.args = {
//   ...HeaderStories.LoggedOut.args,
// };
