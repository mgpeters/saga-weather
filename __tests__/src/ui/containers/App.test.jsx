import React from 'react';
import { shallow } from 'enzyme';

import App from '../../../../src/ui/containers/App.jsx';

describe('App container', () => {
  const wrapper = shallow(<App />);

  it('matches previous snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a 1 section', () => {
    const section = wrapper.find('section');
    expect(section).toHaveLength(1);
  });
});
