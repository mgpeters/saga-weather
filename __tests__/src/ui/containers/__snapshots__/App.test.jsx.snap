// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container matches previous snapshot 1`] = `ShallowWrapper {}`;
